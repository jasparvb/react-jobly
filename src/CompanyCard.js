import React from "react";
import { Link } from "react-router-dom";

function CompanyCard({handle, name, description, logo_url}) {
    return (
        <Link className="Card card" to={`/companies/${handle}`}>
            <div className="card-body">
                <h6 className="card-title d-flex justify-content-between">
                    <span className="text-capitalize">{name}</span>
                    <img src={logo_url} alt={`${name} Logo`} />
                </h6>
                <p>{description}</p>
            </div>
        </Link>
    );
}

export default CompanyCard;