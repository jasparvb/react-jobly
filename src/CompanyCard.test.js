import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("renders without crashing", function() {
    render(
        <MemoryRouter>
        <CompanyCard />
        </MemoryRouter>
    );
});

it("matches snapshot for company", function() {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard 
                handle="walmart" 
                name="Walmart" 
                description="Family Company" 
                logo_url="http://walmart.com/image.jpg" 
            />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});
