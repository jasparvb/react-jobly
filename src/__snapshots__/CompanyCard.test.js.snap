// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for company 1`] = `
<DocumentFragment>
  <a
    class="Card card my-2"
    href="/companies/walmart"
  >
    <div
      class="card-body"
    >
      <h3
        class="card-title d-flex justify-content-between"
      >
        <span
          class="text-capitalize"
        >
          Walmart
        </span>
        <img
          alt="Walmart Logo"
          src="http://walmart.com/image.jpg"
        />
      </h3>
      <p>
        Family Company
      </p>
    </div>
  </a>
</DocumentFragment>
`;
